yearn/contracts/SaddleStrategy.sol

```diff
@@ -0,0 +1,396 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.6.12;
+pragma experimental ABIEncoderV2;
+
+import "@keep-network/yearn-vaults/contracts/BaseStrategy.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/math/Math.sol";
+
+/// @notice Interface for the optional metadata functions from the ERC20
+///         standard.
+interface IERC20Metadata {
+    function symbol() external view returns (string memory);
+}
+
+/// @notice Interface for the Saddle Swap contract.
+/// @dev This is an interface with just a few function signatures of the pool
+///      swap contract. For more info and function description please see:
+///      https://github.com/saddle-finance/saddle-contract/blob/master/contracts/Swap.sol
+///      https://github.com/saddle-finance/saddle-contract/blob/master/contracts/guarded/SwapGuarded.sol
+interface ISaddlePoolSwap {
+    function addLiquidity(
+        uint256[] calldata amounts,
+        uint256 minToMint,
+        uint256 deadline,
+        bytes32[] calldata merkleProof
+    ) external returns (uint256);
+}
+
+/// @notice Interface for the LPRewards contracts.
+/// @dev This is an interface with just a few functions. For more info and
+///      function description please see:
+///      https://github.com/keep-network/keep-ecdsa/blob/main/solidity/contracts/LPRewards.sol
+interface ILPRewards {
```

Comment:
What are your thoughts about adding dependency `"@keep-network/keep-ecdsa": "1.6.0"` in `package.json` for `LPRewards` instead? We have an npm package and we'll be able to import `LPRewards` instead of defining this interface here.

Reply:
`ILPRewards` is an interface for all `LPRewards`-like contracts. In this case, we are talking with `LPRewardsTBTCSaddle` using this interface. The `LPRewardsTBTCSaddle` is for tBTC v1 Saddle pool. For tBTC v2, we will have (probably) a new pool and a new `LPRewards` contract. Because of that, I think it is better to keep a small interface instead of opting for a package dependency.

Reply:
>  For tBTC v2, we will have (probably) a new pool and a new LPRewards contract

Okay, I agree that in this case it's better to stick with an interface.  Maybe I'd add a TODO to verify functions for a new `LPRewards`, cause I take that now we "assume" that this is what it's going to look like.

Reply:
Most likely, it will be just a new instance of the same contract but yeah, we could add a TODO just in case.

Reply:
I added a TODO in: https://github.com/keep-network/tbtc-v2/pull/13/commits/802c155d8e6b78bfe17a77572814367052f3b4f7: